lib/http/2/compressor.rb

```diff
@@ -549,7 +551,16 @@ def header(buf)
       # @return [Array] +[[name, value], ...]+
       def decode(buf)
         list = []
-        list << @cc.process(header(buf)) until buf.empty?
+        decoded_first_regular_header = false
+        until buf.empty?
+          next_header = @cc.process(header(buf))
+          is_pseudo_header = next_header.first.start_with? ':'
+          if decoded_first_regular_header && is_pseudo_header
+            fail CompressionError, 'one or more pseudo headers encountered after regular headers'
```

Comment:
CompressionError is a subclass of ProtocolError, and therefore I raising such an error does adhere to the spec... am I interpret.  Since the error is being raised from compressor.rb, I thought you would have preferred the subclass, CompressionError.  Please, confirm you really want ProtocolError to be raised.

Reply:
Thanks for confirming.